fuel-core/src/schema/tx/types.rs

```diff
@@ -214,8 +219,25 @@ impl Transaction {
 
     async fn status(&self, ctx: &Context<'_>) -> async_graphql::Result<Option<TransactionStatus>> {
         let db = ctx.data_unchecked::<Database>();
-        let status = db.get_tx_status(&self.0.id())?;
-        Ok(status.map(Into::into))
+
+        let tx_pooldb = Box::new(db.clone()) as Box<dyn TxPoolDb>;
+
+        let txpool = TxPoolService::new(tx_pooldb, Arc::new(Config::default()));
```

Comment:
Isn't this creating an brand new instance of a txpool on each status request? We should be fetching the shared txpool out of the `ctx` like we do with the db. The db backs the txpool, but a lot of the data in the txpool is managed in memory, so creating an ephemeral txpool like this will not share any data with the txpool used in other places.

Reply:
I wasn't sure if this was the best way to access the txpool, I went with this approach as I wanted to avoid side effects (hence the clone), but I see why this could be extremely expensive and how the shared data could end up being helpful, so I will move this to fetch from the context

Reply:
The clone is of the db, which is just cloning a pointer and virtually has no cost. The issue is that creating a new txpool instance wouldn't know about the transaction submitted by the user, as it's a different instance. So the transaction they're looking for would never exist in the temp pool you're making here.

Reply:
Ok, will keep that in mind when deciding between whether to clone or pull in an object from the context.

What is the best way to fetch the TxPool from a context - `let txpool = ctx.data::<Arc<dyn TxPool>>().unwrap();` however this results in many tests breaking with `Custom { kind: Other, error: Curl("Send failed since rewinding of the data stream failed"): Send failed since rewinding of the data stream failed }`, what would be the correct way to fetch the TxPool or TxPoolService from the current context?

Reply:
It's injected into the context here: https://github.com/FuelLabs/fuel-core/blob/master/fuel-core/src/service/graph_api.rs#L37

So based on the type use there it would be the concrete `Arc<TxPool>` instead of dyn TxPool.